### PR TITLE
Docs: Improve style documentation and mark the cuppertino style as experimental

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

File: docs/reference/src/advanced/style.md [hunk 1]
```diff
@@ -1,54 +1,53 @@
...
-Before you start your compilation, you can select the style by setting the `SLINT_STYLE` variable
-to one of the style names, such as `fluent` for example.
+-   `fluent`, `fluent-light`, `fluent-dark`: These variants belong to the **Fluent** style, which is based on the [Fluent Design System](https://fluent2.microsoft.design/).
+-   `material`, `material-light`, `material-dark`: These variants are part of the **Material** style, which follows the [Material Design](https://m3.material.io).
+-   `cupertino`, `cupertino-light`, `cupertino-dark`: The **Cupertino** variants emulate the style used by macOS applications. (Experimental)
+-   `native`: The **Native** style mimics the controls native to the platform on which they are used. This includes specific support for the appearance of controls on macOS and Windows. This style requires Qt to be installed on your system.
 
-## Using the `slint_build` Crate
+Some styles have a `-light` and `-dark` variant, in addition to their standard variant.
+The standard variant will switch between the dark or light variant based on your desktop settings.
```
Comment (tronical) [inline docs/reference/src/advanced/style.md:14]: I suggest to flip the order around:

```suggestion
By default, the styles automatically adapt to the system's dark or light color setting. Select a `-light` or `-dark` variant to override the system setting and always show either dark or light colors.
```

File: docs/reference/src/advanced/style.md [hunk 2]
```diff
@@ -1,54 +1,53 @@
...
+If no style is selected, `native` becomes the default if Qt is installed. If `native` isn't available, `fluent` is selected as the default.
 
-Select the style with the [`slint_interpreter::ComponentCompiler::set_style()`](https://docs.rs/slint-interpreter/newest/slint_interpreter/struct.ComponentCompiler.html#method.set_style) function.
 
-</details>
+## Selecting a Widget Style with Rust:
 
-<details data-snippet-language="cpp">
-<summary>Selecting a Widget Style when using Slint with C++:</summary>
+You can choose the style before starting your compilation by setting the `SLINT_STYLE` variable to the name of your chosen style.
```
Comment (tronical) [inline docs/reference/src/advanced/style.md:23]:
```suggestion
You can select the style before starting your compilation by setting the `SLINT_STYLE` environment variable to the name of your chosen style.
```

File: docs/reference/src/advanced/style.md [hunk 3]
```diff
@@ -1,54 +1,53 @@
...
 
-Select the style by defining a `SLINT_STYLE` CMake cache variable to hold the style name as a string. This can be done for example on the command line:
+When using the `slint_build` API, utilize the [`slint_build::compile_with_config()`](https://docs.rs/slint-build/newest/slint_build/fn.compile_with_config.html) function.
```
Comment (tronical) [inline docs/reference/src/advanced/style.md:25]:
```suggestion
When using the `slint_build` API, call the [`slint_build::compile_with_config()`](https://docs.rs/slint-build/newest/slint_build/fn.compile_with_config.html) function.
```

File: docs/reference/src/advanced/style.md [hunk 4]
```diff
@@ -1,54 +1,53 @@
...
 
-```sh
-cmake -DSLINT_STYLE="material" /path/to/source
-```
+When using the `slint_interpeter` API, employ the [`slint_interpreter::ComponentCompiler::set_style()`](https://docs.rs/slint-interpreter/newest/slint_interpreter/struct.ComponentCompiler.html#method.set_style) function.
```
Comment (tronical) [inline docs/reference/src/advanced/style.md:27]:
```suggestion
When using the `slint_interpeter` API, call the [`slint_interpreter::ComponentCompiler::set_style()`](https://docs.rs/slint-interpreter/newest/slint_interpreter/struct.ComponentCompiler.html#method.set_style) function.
```

File: docs/reference/src/advanced/style.md [hunk 5]
```diff
@@ -1,54 +1,53 @@
...
+
+## Selecting a Widget Style when using C++
 
-</details>
+Define a `SLINT_STYLE` CMake cache variable to contain the style name as a string. This can be done, for instance, on the command line:
+
+cmake -DSLINT_STYLE="material" /path/to/source
```
Comment (tronical) [inline docs/reference/src/advanced/style.md:33]: I don't know how to do this with the inline suggestion, but I think this should be surrounded by a code fence backtick series :)

File: docs/reference/src/advanced/style.md [hunk 6]
```diff
@@ -1,54 +1,53 @@
...
 
 ## Previewing Designs With `slint-viewer`
 
-Select the style either by setting the `SLINT_STYLE` environment variable, or passing the style name with the `--style` argument:
+Choose the style either by setting the `SLINT_STYLE` environment variable, or by passing the style name with the `--style` argument:
```
Comment (tronical) [inline docs/reference/src/advanced/style.md:37]:
```suggestion
Select the style either by setting the `SLINT_STYLE` environment variable, or by passing the style name with the `--style` argument:
```

**Review by tronical — APPROVED** [review]

File: docs/reference/src/advanced/style.md [hunk 7]
```diff
@@ -1,54 +1,53 @@
...
-language you're using Slint with.
+Slint offers a variety of [built-in widgets](../language/widgets/widgets.md) which can be accessed from `"std-widgets.slint"`. You can modify the look of these widgets by choosing a style.
 
-<details data-snippet-language="rust">
-<summary>Selecting a Widget Style when using Slint with Rust:</summary>
+The styles available include:
 
-Before you start your compilation, you can select the style by setting the `SLINT_STYLE` variable
-to one of the style names, such as `fluent` for example.
+-   `fluent`, `fluent-light`, `fluent-dark`: These variants belong to the **Fluent** style, which is based on the [Fluent Design System](https://fluent2.microsoft.design/).
```
[the remark this answers is not in the document]
Comment (ogoffart) [reply docs/reference/src/advanced/style.md:8]: Yes, that's probably right.
What's valid syntax for table i can use here?

File: docs/reference/src/advanced/style.md (hunk 5 above)
Comment (ogoffart) [reply docs/reference/src/advanced/style.md:33]: You can use more backticks to escape: 
````suggestion
```sh
cmake -DSLINT_STYLE="material" /path/to/source
```
````

File: docs/reference/src/advanced/style.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (tronical) [reply docs/reference/src/advanced/style.md:8]: Markdown table should be fine AFAIK

**Pull request merged**